web3/exceptions.py

```diff
@@ -58,108 +73,108 @@ def __str__(self):
         return self.args[0]
 
 
-class MismatchedABI(Exception):
+class MismatchedABI(Exception, Web3Exception):
     """
     Raised when an ABI does not match with supplied parameters, or when an
     attempt is made to access a function/event that does not exist in the ABI.
     """
     pass
 
 
-class FallbackNotFound(Exception):
+class FallbackNotFound(Exception, Web3Exception):
     """
     Raised when fallback function doesn't exist in contract.
     """
     pass
 
 
-class ValidationError(Exception):
+class ValidationError(Exception, Web3Exception):
     """
     Raised when a supplied value is invalid.
     """
     pass
 
 
-class NoABIFunctionsFound(AttributeError):
+class NoABIFunctionsFound(AttributeError, Web3Exception):
     """
     Raised when an ABI is present, but doesn't contain any functions.
     """
     pass
 
 
-class NoABIFound(AttributeError):
+class NoABIFound(AttributeError, Web3Exception):
```

Comment:
@kclowes can you make a note in your breaking v6 changes tracking issue to remove `AttributeError` from these exception classes bases.